src/blocks/privacy.rs

```diff
@@ -0,0 +1,246 @@
+//!
+//! # vl4 Options
+//!
+//! Key | Values | Required | Default
+//! ----|--------|----------|--------
+//! `name` | `vl4` | Yes | None
+//! `exclude_device` | A device to ignore, example: `["/dev/video5"]` | No | `[]`
+//! `exclude_consumer` | Processes to ignore | No | `["pipewire", "wireplumber"]`
+//!
+//! # Available Format Keys
+//!
+//! Placeholder                                      | Value                                          | Type     | Unit
+//! -------------------------------------------------|------------------------------------------------|----------|-----
+//! `icon_{audio,audio_sink,video,webcam,unknown}`   | A static icon                                  | Icon     | -
+//! `info_{audio,audio_sink,video,webcam,unknown}`   | The mapping of which source are being consumed | Text     | -
+//!
+//! You can use the suffixes noted above to get the following:
+//!
+//! Suffix       | Description
+//! -------------|------------
+//! `audio`      | Captured audio (ex. Mic)
+//! `audio_sink` | Audio captured from a sink (ex. openrgb)
+//! `video`      | Video capture (ex. screen capture)
+//! `webcam`     | Webcam capture
+//! `unknown`    | Anything else
+//!
+//! # Available Actions
+//!
+//! Action          | Description                               | Default button
+//! ----------------|-------------------------------------------|---------------
+//! `toggle_format` | Toggles between `format` and `format_alt` | Left
+//!
+//! # Example
+//!
+//! ```toml
+//! [[block]]
+//! block = "privacy"
+//! [[block.driver]]
+//! name = "v4l"
+//! [[block.driver]]
+//! name = "pipewire"
+//! exclude_input = ["openrgb"]
+//! display = "nickname"
+//! ```
+//!
+//! # Icons Used
+//! - `microphone`
+//! - `volume`
+//! - `xrandr`
+//! - `webcam`
+//! - `unknown`
+
+use std::collections::HashSet;
+
+use futures::future::{select_all, try_join_all};
+
+use super::prelude::*;
+
+make_log_macro!(debug, "privacy");
+
+#[cfg(feature = "pipewire")]
+mod pipewire;
+mod v4l;
+
+#[derive(Deserialize, Debug)]
+#[serde(deny_unknown_fields)]
+pub struct Config {
+    #[serde(default)]
+    pub format: FormatConfig,
+    #[serde(default)]
+    pub format_alt: FormatConfig,
+    pub driver: Vec<PrivacyDriver>,
+}
+
+#[cfg(feature = "pipewire")]
+#[derive(Deserialize, Debug)]
+#[serde(tag = "name", rename_all = "snake_case")]
+pub enum PrivacyDriver {
+    Pipewire(pipewire::Config),
+    V4l(v4l::Config),
+}
+
+#[cfg(not(feature = "pipewire"))]
+#[derive(Deserialize, Debug)]
+#[serde(tag = "name", rename_all = "snake_case")]
+pub enum PrivacyDriver {
+    V4l(v4l::Config),
+}
+
+#[derive(Debug, Clone, Eq, Hash, PartialEq)]
+enum Type {
+    Audio,
+    AudioSink,
+    Video,
+    Webcam,
+    Unknown,
+}
+
+// {type: {name: {reader}}
+type PrivacyInfo = HashMap<Type, PrivacyInfoInner>;
+
+type PrivacyInfoInnerType = HashMap<String, HashSet<String>>;
```

Comment:
Using a hashset here may not be the best, maybe use a HashMap<String, usize> to be able to give a count perhaps you'd see

``` mic_icon { sink -> [Google Chrome input]}```
or 
``` mic_icon { sink -> [Google Chrome input (x2)]}```

Reply:
Sounds like a good idea